src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web.UnitTests/Controllers/DraftApprenticeshipControllerTests/WhenSelectingCourseOnEditApprenticeship.cs

```diff
@@ -0,0 +1,124 @@
+        {
+            var fixture = new WhenSelectingCourseOnEditApprenticeshipFixture()
+                .WithDraftApprenticeship();
+
+            var result = await fixture.Sut.SelectCourseForEdit(fixture.Request);
+            result.VerifyReturnsViewModel().ViewName.Should().Be("SelectCourse");
+            var model = result.VerifyReturnsViewModel().WithModel<SelectCourseViewModel>();
+            model.CourseCode.Should().Be(fixture.DraftApprenticeship.CourseCode);
+            model.Courses.Should().BeEquivalentTo(fixture.TrainingCourseResponse.TrainingCourses);
+        }
+
+        [Test]
+        public async Task WhenSettingCourse_AndNoCourseSelected_ShouldThrowException()
+        {
+            var fixture = new WhenSelectingCourseOnEditApprenticeshipFixture();
+            fixture.ViewModel.CourseCode = null;
+
+            try
+            {
+                var result = await fixture.Sut.SetCourseForEdit(fixture.ViewModel);
+                Assert.Fail("Should have had exception thrown");
+            }
+            catch (CommitmentsApiModelException e)
+            {
+                e.Errors[0].Field.Should().Be("CourseCode");
+                e.Errors[0].Message.Should().Be("Please select a course");
+            }
+        }
+
+        [Test]
+        public async Task WhenSettingCourse_AndCourseSelected_ShouldRedirectToEditDraftApprenticeship()
+        {
+            var fixture = new WhenSelectingCourseOnEditApprenticeshipFixture()
+                .WithDraftApprenticeship();
+
+            fixture.ViewModel.CourseCode = "123";
+
+            var result = await fixture.Sut.SetCourseForEdit(fixture.ViewModel);
+            result.VerifyReturnsRedirectToActionResult().ActionName.Should().Be("SelectDeliveryModelForEdit");
+        }
+    }
+
+    public class WhenSelectingCourseOnEditApprenticeshipFixture
+    {
+        public DraftApprenticeshipController Sut { get; set; }
+
+        public string RedirectUrl;
+        public Mock<IModelMapper> ModelMapperMock;
+        public Mock<IAuthorizationService> AuthorizationServiceMock;
+        public Mock<ITempDataDictionary> TempDataMock;
+        public Mock<IMediator> MediatorMock;
+        public SelectCourseViewModel ViewModel;
+        public DraftApprenticeshipRequest Request;
+        public EditDraftApprenticeshipViewModel DraftApprenticeship;
+        public GetCohortResponse Cohort;
+        public GetTrainingCoursesQueryResponse TrainingCourseResponse;
+        public Mock<ICommitmentsApiClient> CommitmentsApiClientMock;
+
+        public WhenSelectingCourseOnEditApprenticeshipFixture()
+        {
+            var fixture = new Fixture();
+            ViewModel = fixture.Create<SelectCourseViewModel>();
+            Request = fixture.Create<DraftApprenticeshipRequest>();
+            DraftApprenticeship = fixture.Build<EditDraftApprenticeshipViewModel>().Without(x => x.BirthDay).Without(x => x.BirthMonth).Without(x => x.BirthYear)
```

Comment:
Is it important to the tests that these properties are not set?

Reply:
Autofixture, creates them with invalid values, if not excluded